cmd/thanos/sidecar.go

```diff
@@ -160,20 +142,13 @@ func runSidecar(
 			}
 
 			// New gossip cluster.
-			err = peer.Join(
-				cluster.PeerState{
-					Type:    cluster.PeerTypeSource,
-					APIAddr: grpcAddr,
-					Metadata: cluster.PeerMetadata{
-						Labels: externalLabels.GetPB(),
-						// Start out with the full time range. The shipper will constrain it later.
-						// TODO(fabxc): minimum timestamp is never adjusted if shipping is disabled.
-						MinTime: 0,
-						MaxTime: math.MaxInt64,
-					},
-				},
-			)
-			if err != nil {
+			if err = peer.Join(cluster.PeerTypeSource, cluster.PeerMetadata{
```

Comment:
this seems to be in a few places ... not saying do now but in future might be good to consolidate

Reply:
Consolidate what? it is just `Join` method with different parameters for different commands. Not sure how to consolidate.